backend/internal/controller/adapter/dispatcher.go

```diff
@@ -0,0 +1,68 @@
+package adapter
+
+import (
+	"encoding/json"
+	"io"
+)
+
+type Category int
+
+const (
+	None Category = iota
+	Initial
+	Goal
+	Position
```

Comment:
Maybe names could have some prefixes, to better represent their purpose? Currently, they read `adapter.Initial`, which might be a bit misleading